sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/PartitionOwnership.cs

```diff
@@ -68,7 +68,7 @@ public class PartitionOwnership
         ///   The entity tag needed to update this ownership.
         /// </summary>
         ///
-        public string ETag { get; }
+        public string ETag { get; set; }
```

Comment:
Should this be an `internal` set?   Do we want external developers responsible for ETags?

Reply:
Personally, I'd make this a calculated field; rather than having something external responsible for the eTag, this class computes it on demand as some kind of a hash of it's values or whatever approach.

Reply:
The set is public because users need to modify the `ETag` before persisting it into their storage service.

The flow goes as follows:
- The SDK decides to claim ownership of a partition (this should happen in load balancing, not implemented yet).
- The SDK creates a new `PartitionOwnership` instance (which has an internal constructor) with an `ETag` that matches the existing ownership (so it can be claimed).
- The `PartitionOwnership` instance is passed to the `PartitionManager` (user code) when calling the `ClaimOwnership` method.
- The user code verifies whether the `ETag` matches or not. If it matches, it stores the `PartitionOwnership` into their storage service with **a newly created ETag**. So the user must be able to modify it.

That's exactly what we are doing in `InMemoryPartitionManager`. It's a `PartitionManager` subclass provided by us, but it should be able to be implemented by any user, so it only makes use of public stuff.

Reply:
Given that the `ETag` value is always just some random stuff (that's what I'm assuming), I think your suggestion is reasonable. Maybe we should make the `set` internal and create a method to modify the `ETag` (like `void UpdateETag(void)` or something like that)?

Reply:
We should defer the conversation until after preview two, but I think that delegating responsibility here is the wrong decision.   Personally, I'd compute the eTag dynamically, on demand, based on the members of the class.  While, technically, the eTag just needs to be a unique value that represents the state, it is often helpful to have a deterministic value that represents the actual state such that two instances with the same state would have the same value.